### PR TITLE
Adding last approved avatar date to mod avatar approval page

### PR DESCRIPTION
**Pull request by kev306**

### CODE REVIEW

File: src/databaseAdapters/user.ts [hunk 1]
```diff
@@ -33,21 +30,20 @@ class MongoUserAdapter implements DatabaseAdapter {
...
 
-  async setAvatarAndUpdateLibrary(username: string, avatarSet: S3AvatarSet) {
+  async approveAvatar(username: string, avatarSet: S3AvatarSet) {
     const user = await this.getUser(username);
-
-    user.avatarImgRes = avatarSet.resLink;
-    user.avatarImgSpy = avatarSet.spyLink;
-    user.avatarLibrary.push(avatarSet.avatarSetId);
-
     const librarySize = await getAvatarLibrarySizeForUser(
```
Comment (vck3000) [inline src/databaseAdapters/user.ts:35]: On second thought, I don't think this guy should be here. It should be one level higher in the caller.

File: src/routes/profile/index.tsx [hunk 2]
```diff
@@ -66,13 +68,44 @@ router.get('/avatargetlinktutorial', (req, res) => {
...
+          forUsername: avatarReq.forUsername,
+          resLink: avatarReq.resLink,
+          spyLink: avatarReq.spyLink,
+          lastApprovedAvatarDate: user.lastApprovedAvatarDate
+            ? user.lastApprovedAvatarDate
+            : null,
+          enoughTimeElapsed: user.lastApprovedAvatarDate
+            ? user.lastApprovedAvatarDate <
+              new Date() - MIN_TIME_SINCE_LAST_AVATAR_APPROVAL
+            : true,
```
Comment (vck3000) [inline src/routes/profile/index.tsx:99]: Add in library size check (should not show red to mods if they are currently at 1 but they have lib size 2). And think of a better name.

**Comment by kev306** [comment]
Added notification message if avatar library is full

image

**Comment by kev306** [comment]
Added checklist + date format:

Happy path:
image

Validation checks failed:
image

**Comment by kev306** [comment]
image

File: src/databaseAdapters/user.ts [hunk 3]
```diff
@@ -33,17 +34,18 @@ class MongoUserAdapter implements DatabaseAdapter {
...
+    librarySize: number,
+  ): Promise<void> {
     const user = await this.getUser(username);
 
+    user.lastApprovedAvatarDate = new Date();
     user.avatarImgRes = avatarSet.resLink;
     user.avatarImgSpy = avatarSet.spyLink;
     user.avatarLibrary.push(avatarSet.avatarSetId);
 
-    const librarySize = await getAvatarLibrarySizeForUser(
```
Comment (vck3000) [inline src/databaseAdapters/user.ts:43]: Remove the import pls :)

File: src/routes/profile/index.tsx [hunk 4]
```diff
@@ -66,13 +68,54 @@ router.get('/avatargetlinktutorial', (req, res) => {
```
Comment (vck3000) [inline src/routes/profile/index.tsx:68]: Can u check this guy avatargetlinktutorial
and remove it if it's unused lol

**Review by vck3000 — APPROVED** [review]
LGTM 🚢

**Pull request merged**